### ISSUE TITLE
VelopackApp check should handle `async Task Main()`

### ISSUE THREAD

**Issue by Arcalise08** [velopack/velopack#55]
Unsure what has changed but I am getting the error on VPK that it cant find my `Velopack.VelopackApp.Build().Run();` statement which is the very first thing Main does. It has been working so i'm unsure why i'm having issues now. I found this code that looks for the run statement from the IL code. Debugging it i see that theres only 7 instructions (which isnt true, i have around 97 instructions) 

![image](https://github.com/velopack/velopack/assets/61042848/b2b84d04-9d2e-4a2e-bbb5-1c7ed9fec4d1)
With async main it loads these instructions
![image](https://github.com/velopack/velopack/assets/61042848/b7a8517d-d32e-496d-a138-46ad02e3a1bf)

Commenting out the async/await and making main non-async I properly load all instructions
![image](https://github.com/velopack/velopack/assets/61042848/712c9ffd-cb04-49cb-9551-f944f2f0e74b)
![image](https://github.com/velopack/velopack/assets/61042848/ede19281-d42b-490d-a907-52ef913d388f)

Heres offending code in velopack

```
            foreach (var instr in entryPoint.CilMethodBody.Instructions) {
                if (instr.OpCode.Code is CilCode.Call or CilCode.Callvirt or CilCode.Calli) {
                    SerializedMemberReference operand = instr.Operand as SerializedMemberReference;
                    if (operand != null && operand.IsMethod) {
                        if (operand.Name == "Run" && operand.DeclaringType.FullName == "Velopack.VelopackApp") {
                            // success!
                            if (velopackVersion != null) {
                                log.Info($"Verified VelopackApp.Run() in '{entryPoint.FullName}', version {velopackVersion}.");
                                if (velopackVersion != VelopackRuntimeInfo.VelopackProductVersion) {
                                    log.Warn(exeFile + " was built with a different version of Velopack than this tool. " +
                                        $"This may cause compatibility issues. Expected {VelopackRuntimeInfo.VelopackProductVersion}, " +
                                        $"but found {velopackVersion}.");
                                }
                                return velopackVersion;
                            } else {
                                log.Warn("VelopackApp verified at entry point, but ProductVersion could not be checked.");
                                return null;
                            }
                        }
                    }
                }
            }
```

**Comment by caesay** [comment]
This makes sense, "async Main" is not real, you can see from the instructions loaded that there is a real non-async Main method which basically calls `return AsyncMain().GetAwaiter().GetResult()`. I think we could improve this check though to handle this case. In the mean time, you can turn off this check with the `--skipVeloAppCheck` argument

**Comment by Arcalise08** [comment]
I do recognize that the async main isnt "real" but just something helpful that MS has added to make it easier to use async at the entry of a program. I believe a mention on the docs would be helpful. Most of the templates for dotnet apps are using inline Main as well so it might be worth the time investment to make it work seamlessly.

**Comment by caesay** [comment]
Fixed in f1fd069ac7c7482bbdcadc48bdca2bca20e90349 - we will now allow VelopackApp.Run() to exist in any method within the main module, rather than just the ManagedEntryPoint.
